### PR TITLE
refactor: use modal steps wrapper for tracking events

### PR DESCRIPTION
**Pull request by el**
## Changes

Great work from @rebelchris for simplifying this

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/components/onboarding/ThemeOnboarding.tsx [hunk 1]
```diff
@@ -35,16 +29,10 @@ function ThemeOnboarding({ onClose }: OnboardingStepProps): ReactElement {
             ))}
           </Container>
           <Modal.Footer justify={Justify.Between}>
-            <Button
-              className="btn-tertiary"
-              onClick={onStepBackward(previousStep)}
-            >
-              {index === 1 ? 'Close' : 'Back'}
+            <Button className="btn-tertiary" onClick={previousStep}>
+              {activeStepIndex === 1 ? 'Close' : 'Back'}
```
Comment (sshanzel) [inline packages/shared/src/components/onboarding/ThemeOnboarding.tsx:33]: If this button will bring the user back to the introduction, with the removal of `onClose`, then we can just keep the label as `Back`.

**Review by sshanzel — APPROVED** [review]
Just one concern with the label to display when allowing to go back to the intro, other than that, everything looks good to me! 🚢

**Review by rebelchris — APPROVED** [review]
We do have to ensure all possible flows still work for this (intro + any form of steps)
And verify the analytics get triggered.

Code wise happy with the changes.

File: packages/shared/src/components/onboarding/ThemeOnboarding.tsx (hunk 1 above)
Comment (el) [reply packages/shared/src/components/onboarding/ThemeOnboarding.tsx:33]: Great catch

File: packages/shared/src/contexts/OnboardingContext.tsx [hunk 2]
```diff
@@ -55,7 +55,7 @@ export const OnboardingContextProvider = ({
   const { loadedAlerts, alerts } = useContext(AlertContext);
   const { trackEvent } = useContext(AnalyticsContext);
   const { registerLocalFilters } = useMyFeed();
-  const [isOnboarding, setIsOnboarding] = useState(false);
```
Comment (el) [inline packages/shared/src/contexts/OnboardingContext.tsx:58]: Fixed here https://github.com/dailydotdev/apps/pull/1607/files

**Pull request merged**